{
  ExceptionalSupplier<String,IOException> task=createMock(new Clazz<ExceptionalSupplier<String,IOException>>(){
  }
);
  IOException thrown=new IOException();
  expect(task.get()).andThrow(thrown);
  control.replay();
  try {
    backoffHelper.doUntilResult(task);
    fail("Expected exception to be bubbled");
  }
 catch (  IOException e) {
    assertSame(thrown,e);
  }
  control.verify();
}
